src/sentry/static/sentry/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -76,9 +80,46 @@ class OrganizationIntegrations extends AsyncComponent<
     organization: SentryTypes.Organization,
   };
 
+  getDefaultState() {
+    return {
+      ...super.getDefaultState(),
+      list: [],
+      displayedList: [],
+    };
+  }
+
   onLoadAllEndpointsSuccess() {
+    const {integrations, publishedApps, orgOwnedApps, extraApp, plugins} = this.state;
+    const published = publishedApps || [];
+    // If we have an extra app in state from query parameter, add it as org owned app
+    if (extraApp) {
+      orgOwnedApps.push(extraApp);
+    }
+
+    // we dont want the app to render twice if its the org that created
+    // the published app.
+    const orgOwned = orgOwnedApps.filter(app => {
+      return !published.find(p => p.slug === app.slug);
+    });
+
+    /**
+     * We should have three sections:
+     * 1. Public apps and integrations available to everyone
+     * 2. Unpublished apps available to that org
+     * 3. Internal apps available to that org
+     */
+
+    const combined = ([] as AppOrProviderOrPlugin[])
+      .concat(published)
+      .concat(orgOwned.filter(a => a.status === 'published'))
+      .concat(this.providers)
+      .concat(plugins);
+
+    const list = this.sortIntegrations(combined);
+
+    this.setState({list, displayedList: list});
+
     //count the number of installed apps
-    const {integrations, publishedApps} = this.state;
```

Comment:
@NisanthanNanthakumar Might be worth putting the analytics into a separate function so `onLoadAllEndpointsSuccess` isn't as big.